promql/parser/prettier_rules.md

```diff
@@ -0,0 +1,31 @@
+# Prettifying PromQL expressions
+This files contains rules for prettifying PromQL expressions.
+
+### Keywords
+`max_characters_per_line`: Maximum number of characters that will be allowed in a single stringified node
+
+## Rules
+1. A node exceeding the `max_characters_per_line` will qualify for split unless
+   1. It is a function call with one argument eg. `rate(foo[1m])`
```

Comment:
What if the argument is big? Do we not split in that case?

Reply:
No, based on https://github.com/prometheus/prometheus/pull/10544#discussion_r880236631, we do not split if a call is with 1 argument.

Reply:
I feel it should be broken into multiple lines. Because that one argument can be very big. For example here https://github.com/prometheus/prometheus/pull/10544#discussion_r880310693, it should be broken into multiple lines for easy reading